integration_test.go

```diff
@@ -571,3 +596,82 @@ func TestPort(t *testing.T) {
 		t.Fatalf("expected health to be 0, but is %d", deploys[0].Health)
 	}
 }
+
+func TestCleanup(t *testing.T) {
+	client, server := startCamus(t)
+	defer server.Kill()
+	defer client.Shutdown()
+	client.Build()
+
+	// deploy without running and then cleanup
+	deployOnly := client.Push()
+	if !findDeployById(deployOnly, client.ListDeploys()) {
+		t.Fatal("newly minted deploy should be in the new deploy list")
+	}
+	client.Cleanup(deployOnly)
+	if findDeployById(deployOnly, client.ListDeploys()) {
+		t.Fatal("deploy should have been cleaned up")
+	}
+
+	// deploy, run then attempt cleanup. then stop and cleanup
+	deployStop := client.Push()
+	client.Run(deployStop)
+	if err := client.client.CleanupDeploy(deployStop); err == nil {
+		t.Fatal("expected error when cleaning up a running deploy")
+	}
+	if !findDeployById(deployStop, client.ListDeploys()) {
+		t.Fatal("deployStop should still be there after failed cleanup")
+	}
+	client.Stop(deployStop)
+	client.Cleanup(deployStop)
+	if findDeployById(deployStop, client.ListDeploys()) {
+		t.Fatal("deploy should have been cleaned up")
+	}
+
+	// cleanup something non-existent
+	if err := client.client.CleanupDeploy("non-existent-deploy"); err == nil {
+		t.Fatal("expected error when cleaning up a non-existent-deploy")
+	}
+
+	// deploy, run then attempt cleanup. then kill process and cleanup
+	deployKill := client.Push()
+	client.Run(deployKill)
+	if err := client.client.CleanupDeploy(deployKill); err == nil {
+		t.Fatal("expected error when cleaning up a running deploy")
+	}
+	if !findDeployById(deployKill, client.ListDeploys()) {
+		t.Fatal("deployKill should still be there after failed cleanup")
+	}
+	pid := findDeployPid(deployKill, client.ListDeploys())
+	if pid == 0 {
+		t.Fatal("could not find running process to kill")
+	}
+	// FindProcess never returns error on unix. no point checking
+	proc, _ := os.FindProcess(pid)
+	proc.Kill()
+	client.Cleanup(deployKill)
+	if findDeployById(deployKill, client.ListDeploys()) {
+		t.Fatal("deploy should have been cleaned up")
+	}
+
+	// deploy then cleanup and then do a new deploy to see if it runs on the same port
+	// it should run on the same port since it should be the lowest free port number
+	deployA := client.Push()
```

Comment:
I don't think this test belongs here. In fact, cleanup should only be concerned with removing deploys that aren't currently running.

The fact that new deploys start on the lowest free port is an implementation detail.